lib/api/element-commands/_waitFor.js

```diff
@@ -24,7 +26,7 @@ class WaitForElement extends ElementCommand {
   }
 
   validateArgsCount() {
-    if (Utils.isString(this.args[0]) && (Utils.isString(this.args[1]) || ElementCommand.isSelectorObject(this.args[1]))) {
+    if (Utils.isString(this.args[0]) && LocateStrategy.isValid(this.args[0])) {
```

Comment:
the `isSelectorObject` check is lost here

Reply:
I guess that won't be need here as we only want to set strategy from args when a valid strategy is found. Otherwise setting strategy from selector is already taken into consideration [here](https://github.com/nightwatchjs/nightwatch/blob/f14c987213b7a4a5c92e78f546380a943c53f8a2/lib/element/command.js#L137)

Reply:
It necessary in order to make sure the second argument is either a string or a valid selector object. This would prevent incorrect usage scenarios, such as: 
```
.waitForElementVisible('xpath', {})
.waitForElementVisible('css selector', 100, 'test message')
```

But you can simply modify the check to be if:
`(LocateStrategy.isValid(this.args[0]) && (Utils.isString(this.args[1]) || 
ElementCommand.isSelectorObject(this.args[1])))`. 

Also it seems other element commands (like getText, getValue, etc.) could be affected since they dont validate the strategy either; see https://github.com/nightwatchjs/nightwatch/blob/main/lib/element/command.js#L261